Turn macOS smooth keyboard scrolling on by default

rr-codes opened the pull request:
#### cd1dbd2ab3f6005242c7ba9db7356e538d09ad2d
<pre>
Turn macOS smooth keyboard scrolling on by default
<a href="https://bugs.webkit.org/show_bug.cgi?id=228159">https://bugs.webkit.org/show_bug.cgi?id=228159</a>
rdar://80912063

Reviewed by Tim Horton and Wenson Hsieh.

No tests added.

Set the default value for EventHandlerDrivenSmoothKeyboardScrollingEnabled to true in WebKit.

* LayoutTests/css3/scroll-snap/scroll-padding-overflow-paging.html:
Updated to reflect non-instananeous scrolling.

* LayoutTests/fast/repaint/resources/fixed-move-after-keyboard-scroll-iframe.html:
Updated to reflect non-instananeous scrolling.

* LayoutTests/fast/scrolling/arrow-key-scroll-in-rtl-document-expected.txt:
* LayoutTests/fast/scrolling/arrow-key-scroll-in-rtl-document.html:
* LayoutTests/platform/mac-wk1/fast/scrolling/arrow-key-scroll-in-rtl-document-expected.txt: Added.
Updated these tests to reflect the new distance scrolled.

* Source/WTF/Scripts/Preferences/WebPreferencesInternal.yaml:
`EventHandlerDrivenSmoothKeyboardScrollingEnabled` is now true on default for WebKit.

* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::startKeyboardScrollAnimationOnEnclosingScrollableContainer):
Added check if renderer is null to fix crash in LayoutTests/fast/forms/select/select-change-type-on-focus.

* Tools/TestWebKitAPI/Tests/WebKit/SpacebarScrolling.cpp:
(TestWebKitAPI::didRunJavascript):
(TestWebKitAPI::TEST):
Updated to reflect non-instananeous scrolling.

Canonical link: <a href="https://commits.webkit.org/255031@main">https://commits.webkit.org/255031@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2e5a42ee80b60cd4021af0a9db95f4c7d181ae90

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/91043 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/27 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/21571 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/100462 "Built successfully") | [❌ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/159410 "Found 1 new test failure: fast/workers/worker-crash-with-invalid-location.html (failure)") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/95049 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/65 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/29109 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/83398 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/97168 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/96699 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/54 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/77819 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/27004 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/82005 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/81642 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/70041 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/82539 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/35192 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/15664 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/77545 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/32989 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/16654 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/26711 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3501 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/36771 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/39620 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/80143 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/38698 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/35738 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/17558 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on LayoutTests/css3/scroll-snap/scroll-padding-overflow-paging.html:
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html> <!-- webkit-test-runner [ ScrollAnimatorEnabled=false ] -->
+<!DOCTYPE html> <!-- webkit-test-runner [ ScrollAnimatorEnabled=true ] -->
```
I think you can just delete the `webkit-test-runner` header.

hortont424 commented on LayoutTests/fast/repaint/fixed-move-after-keyboard-scroll.html:
```diff
@@ -5,7 +5,7 @@
       {
           if (window.internals)
-              internals.settings.setScrollAnimatorEnabled(false);
+              internals.settings.setScrollAnimatorEnabled(true);
```
This test was *specifically* about the ScrollAnimatorEnabled setting, so it seems wrong to be changing this. Maybe disable animated keyboard scrolling in a test header instead?

hortont424 commented on LayoutTests/fast/scrolling/arrow-key-scroll-in-rtl-document.html:
```diff
@@ -23,17 +23,10 @@
         }
 
-        if (window.internals)
-            internals.settings.setScrollAnimatorEnabled(false);
```
Ditto the previous ScrollAnimatorEnabled thing

hortont424 commented on Source/WTF/Scripts/Preferences/WebPreferencesInternal.yaml:
```diff
@@ -275,7 +275,7 @@ EventHandlerDrivenSmoothKeyboardScrollingEnabled:
    default: false
   WebKit:
-   default: false
+   default: true
```
*dances*

whsieh commented on Tools/TestWebKitAPI/Tests/WebKit/SpacebarScrolling.cpp:
```diff
@@ -49,6 +52,16 @@ static void didNotHandleKeyEventCallback(WKPageRef, WKNativeEventPtr event, cons
+
+static void didRunJavascript(WKSerializedScriptValueRef serializedScriptValue, WKErrorRef error, void* context)
+{
+    JSGlobalContextRef scriptContext = JSGlobalContextCreate(0);
```
Nit - it looks like this leaks `scriptContext`. Do we need a `JSGlobalContextRelease`?

hortont424 commented on LayoutTests/css3/scroll-snap/scroll-padding-overflow-paging.html:
```diff
@@ -1,4 +1,3 @@
-<!DOCTYPE html> <!-- webkit-test-runner [ ScrollAnimatorEnabled=false ] -->
```
Well don't remove the doctype :D

rr-codes committed:
Turn macOS smooth keyboard scrolling on by default
https://bugs.webkit.org/show_bug.cgi?id=228159
rdar://80912063

Reviewed by Tim Horton and Wenson Hsieh.

No tests added.

Set the default value for EventHandlerDrivenSmoothKeyboardScrollingEnabled to true in WebKit.

* LayoutTests/css3/scroll-snap/scroll-padding-overflow-paging.html:
Updated to reflect non-instananeous scrolling.

* LayoutTests/fast/repaint/resources/fixed-move-after-keyboard-scroll-iframe.html:
Updated to reflect non-instananeous scrolling.

* LayoutTests/fast/scrolling/arrow-key-scroll-in-rtl-document-expected.txt:
* LayoutTests/fast/scrolling/arrow-key-scroll-in-rtl-document.html:
* LayoutTests/platform/mac-wk1/fast/scrolling/arrow-key-scroll-in-rtl-document-expected.txt: Added.
Updated these tests to reflect the new distance scrolled.

* Source/WTF/Scripts/Preferences/WebPreferencesInternal.yaml:
`EventHandlerDrivenSmoothKeyboardScrollingEnabled` is now true on default for WebKit.

* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::startKeyboardScrollAnimationOnEnclosingScrollableContainer):
Added check if renderer is null to fix crash in LayoutTests/fast/forms/select/select-change-type-on-focus.

* Tools/TestWebKitAPI/Tests/WebKit/SpacebarScrolling.cpp:
(TestWebKitAPI::didRunJavascript):
(TestWebKitAPI::TEST):
Updated to reflect non-instananeous scrolling.

Canonical link: https://commits.webkit.org/255031@main
